charts/openebs/templates/daemonset-ndm.yaml

```diff
@@ -29,6 +29,10 @@ spec:
         name: openebs-ndm
         openebs.io/version: {{ .Values.release.version }}
     spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
```

Comment:
@sonasingh46 @prateekpandey14  Should we use `with` or `if`.
- We need to be consistent across all the repos.
- We need to be consistent in the templates. For some fields `with` and  for some `if` are used.

Reply:
my two cents.  You should stay with Helm defaults.  If you do helm create test.. you will see that in the deployment file

````
{{- with .Values.imagePullSecrets }}
      imagePullSecrets:
        {{- toYaml . | nindent 8 }}
      {{- end }}
...
{{- with .Values.nodeSelector }}
      nodeSelector:
        {{- toYaml . | nindent 8 }}
      {{- end }}
      {{- with .Values.affinity }}
      affinity:
        {{- toYaml . | nindent 8 }}
      {{- end }}
      {{- with .Values.tolerations }}
      tolerations:
        {{- toYaml . | nindent 8 }}
      {{- end }}
````

Reply:
great. Then I will follow this pattern,